projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/authenticator_factory.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from typing import Union
+
+from vdk.internal.core import errors
+from vdk.plugin.kerberos.base_authenticator import BaseAuthenticator
+from vdk.plugin.kerberos.kerberos_configuration import KerberosPluginConfiguration
+from vdk.plugin.kerberos.kinit_authenticator import KinitGSSAPIAuthenticator
+from vdk.plugin.kerberos.minikerberos_authenticator import (
+    MinikerberosGSSAPIAuthenticator,
+)
+
+log = logging.getLogger(__name__)
+
+
+class KerberosAuthenticatorFactory:
+    def __init__(self, configuration: KerberosPluginConfiguration):
+        self.__configuration = configuration
+
+    def create_authenticator(
+        self, authentication_type: str
+    ) -> Union[BaseAuthenticator, None]:
```

Comment:
Isn't union of instance | None actually an optional instance?

Reply:
It is indeed. I've changed it for consistency.